db/migrate/20220201065329_ensure_os_minor_is_not_nil.rb

```diff
@@ -0,0 +1,12 @@
+class EnsureOsMinorIsNotNil < ActiveRecord::Migration[6.0]
+  def up
+    User.without_auditing do
+      Operatingsystem.unscoped.find_each do |os|
```

Comment:
Can't we filter with `where(minor: nil)` here and update at the DB level?

And I think the DB should be modified to have a NOT NULL, possibly also a DEFAULT empty string.

Reply:
Surprise! We already have the default and NOT NULL in the db, at least this is what I see in my dump:

```
    t.string "name", limit: 255, null: false
    t.string "major", limit: 5, default: "", null: false
    t.string "minor", limit: 16, default: "", null: false
```

Reply:
Oh yeah this was implemented some time ago: https://projects.theforeman.org/issues/297